src/syntaxtree.c

```diff
@@ -306,6 +408,58 @@ static Obj SyntaxTreeIf(Obj result, Stat stat)
     return result;
 }
 
+static Expr SyntaxTreeCodeIf(Obj node)
+{
+    Obj  branches = ElmPRec(node, RNamName("branches"));
+    UInt len = LEN_PLIST(branches);
+    UInt tnum =
+        FIND_TNUM_FROM_STRING(CSTR_STRING(ElmPRec(node, RNamName("type"))));
+    Expr ifexpr = NewStatWithProf(tnum, 2 * len * sizeof(Expr), 0);
+    for (int i = 0; i < len; i++) {
+        Obj  condbodypair = ELM_PLIST(branches, i + 1);
+        Obj  conditionnode = ElmPRec(condbodypair, RNamName("condition"));
+        Obj  bodynode = ElmPRec(condbodypair, RNamName("body"));
+        Expr condition = SyntaxTreeDefaultCoder(conditionnode);
+        Expr body = SyntaxTreeDefaultCoder(bodynode);
+        WRITE_EXPR(ifexpr, 2 * i, condition);
+        WRITE_EXPR(ifexpr, 2 * i + 1, body);
+    }
+    return ifexpr;
+}
+
+static Expr SyntaxTreeCodeImmediateInteger(Obj node)
+{
+    Obj value = ElmPRec(node, RNamName("value"));
+    return INTEXPR_INT(INT_INTOBJ(value));
+}
+
+static Expr SyntaxTreeCodeLargeInteger(Obj node)
+{
+    Obj  value = ElmPRec(node, RNamName("value"));
+    Expr expr = NewStatWithProf(T_INT_EXPR, sizeof(UInt), 0);
+    Int  ix = PushValue(value);
+    WRITE_EXPR(expr, 0, ix);
+    return expr;
+}
+
+static Expr SyntaxTreeCodeString(Obj node)
+{
+    Obj  string = ElmPRec(node, RNamName("string"));
```

Comment:
```suggestion
    Obj  string = ElmPRec(node, RNamName("value"));
```
We renamed `string` to `value` in the decoder. Afterwards, we should be able to unify `SyntaxTreeCodeLargeInteger` and `SyntaxTreeCodeString` by extracting the tnum from `node`, no?

Reply:
Done.